cads_adaptors/adaptors/arco.py

```diff
@@ -36,18 +37,43 @@ def _normalise_location(self, request: Request) -> None:
         except (ValueError, TypeError):
             raise InvalidRequest(f"Invalid {location=}. {msg}")
 
-    def _normalise_date(self, request: Request) -> None:
-        date = ensure_list(request.get("date"))
-        if not date:
-            request["date"] = date
-            return
-
-        if len(date) != 1:
+    def _normalise_date(self, request: Request, date_key="date") -> None:
+        date = ensure_list(request.get(date_key))
+        if len(date) == 1:
+            split = sorted(str(date[0]).split("/"))
+            date_range = [split[0], split[-1]]
+        elif len(date) == 2:
+            date_range = date
+        else:
             raise InvalidRequest(
-                "Please specify a single date range using the format yyyy-mm-dd/yyyy-mm-dd."
+                'Please specify a single date range using the format "yyyy-mm-dd/yyyy-mm-dd" or '
+                '["yyyy-mm-dd", "yyyy-mm-dd"].'
+            )
+
+        # Embargo check
+        if "embargo" in self.config and self.config["embargo"]:
+            from datetime import UTC, datetime, timedelta
+
+            from dateutil.parser import parse as dtparse
+
+            embargo = self.config["embargo"]
+            embargo_error_time_format: str = embargo.pop(
+                "error_time_format", "%Y-%m-%d %H:00"
             )
-        split = sorted(str(date[0]).split("/"))
-        request["date"] = ["/".join([split[0], split[-1]])]
+            embargo_datetime = datetime.now(UTC) - timedelta(**embargo)
+            if dtparse(date_range[0]).date() > embargo_datetime.date():
+                raise InvalidRequest(
+                    "You have requested data under embargo, the latest available data is: "
+                    f" {embargo_datetime.strftime(embargo_error_time_format)}"
+                )
+            if dtparse(date_range[1]).date() > embargo_datetime.date():
+                date_range[1] = embargo_datetime.strftime(embargo_error_time_format)
+                self.context.add_user_visible_error(
+                    "Part of the data you have requested is under embargo, "
+                    "your request has been modified to the latest available data: "
+                    f"{date_key}={date_range}"
+                )
```

Comment:
Most downstream software, including Earthkit, uses local caching. The use of local caching has likely increased now that the CDS cache is at the worker level. As a result, users will permanently cache a request that differs from what they originally asked for.

I know there's a warning, but batch users or API users who do not use `cdsapi` are very likely to miss the warning. Also, I'm not sure the concept of embargo is clear to users.